### ISSUE TITLE
Adding/deleting an edge doesn't update the index of the object

### ISSUE THREAD

**Issue by fredpook** [driebit/mod_elasticsearch#18]
Use case:
A query with: hasobject=1234
shows all rsc's with keyword 1234
When adding a new rsc with keyword 1234 it won't show in the query until a complete re-index.

**Comment by ddeboer** [comment]
I thought this too for a moment, but actually it works perfectly fine: https://github.com/zotonic/zotonic/issues/1698

**Comment by fredpook** [comment]
Editors of Amsterdam Museum seen it happen several times.
Just now I solved 1 issue for them by calling the refresh API explicitly
https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-refresh.html

I didn't find code for refreshing in this module. Should we refresh after each operation?

**Comment by ddeboer** [comment]
No, we don’t need to. There can be a normal delay of ~1 sec between updating some document in Elasticsearch and its being available in the index. 

That delay is normal behaviour, usually not an issue for our use-cases. The problem that some customers are seeing must be caused by some different cause.

**Comment by fredpook** [comment]
"some different cause" in this module since they didn't experience the issues before ...

**Comment by ddeboer** [comment]
Or on the server.

**Comment by fredpook** [comment]
A solution for this kind  of  erratic "out of sync" behaviour could be to have the "simple" hassubject/hasobject queries handled by postgres like before.
These perform very fast in zotonic and don't need ES for improved fulltext search.
Just the combinations of text plus edge queries can be handled by ES

**Comment by fredpook** [comment]
And this https://github.com/zotonic/zotonic/issues/1722 will help too
